govcd/access_control.go

```diff
@@ -456,3 +456,145 @@ func checkSanityVdcControlAccess(vdc *Vdc) error {
 	}
 	return nil
 }
+
+func publishCatalog(client *Client, catalogUrl string, tenantContext *TenantContext, publishCatalog types.PublishCatalogParams) error {
+	catalogUrl = catalogUrl + "/action/publish"
+
+	publishCatalog.Xmlns = types.XMLNamespaceVCloud
+
+	if tenantContext != nil {
+		client.SetCustomHeader(getTenantContextHeader(tenantContext))
+	}
+
+	err := client.ExecuteRequestWithoutResponse(catalogUrl, http.MethodPost,
+		types.PublishCatalog, "error setting catalog publishing state: %s", publishCatalog)
+
+	if tenantContext != nil {
+		client.RemoveProvidedCustomHeaders(getTenantContextHeader(tenantContext))
+	}
+
+	return err
+}
+
+// IsSharedReadOnly returns the state of the catalog read-only sharing to all organizations
+func (cat *Catalog) IsSharedReadOnly() (bool, error) {
+	accessControl, err := cat.GetAccessControl(true)
+	if err != nil {
+		return false, err
+	}
+	if accessControl.AccessSettings != nil || accessControl.IsSharedToEveryone {
+		return false, nil
+	}
+	err = cat.Refresh()
+	if err != nil {
+		return false, err
+	}
+	return cat.Catalog.IsPublished, nil
+}
+
+// IsSharedReadOnly returns the state of the catalog read-only sharing to all organizations
+func (cat *AdminCatalog) IsSharedReadOnly() (bool, error) {
+	accessControl, err := cat.GetAccessControl(true)
+	if err != nil {
+		return false, err
+	}
+	if accessControl.AccessSettings != nil || accessControl.IsSharedToEveryone {
+		return false, nil
+	}
+	err = cat.Refresh()
+	if err != nil {
+		return false, err
+	}
+	return cat.AdminCatalog.IsPublished, nil
+}
+
+// publish publishes a catalog read-only access control to all organizations
```

Comment:
Nit, separator needed for two sentences:

```suggestion
// publish publishes a catalog read-only access control to all organizations.
```